### ISSUE TITLE
Is treating names as case-sensitive a bug?

### ISSUE THREAD

**Issue by ekweirui** [nus-cs2103-AY2223S1/forum#389]
The module website gives an example stating how person names should be case-insensitive, and failure to do so can be considered a feature flaw.
<img width="409" alt="image" src="https://user-images.githubusercontent.com/95850532/198816587-6d4b4b09-76d3-4590-8767-c9d5444b5b9c.png">

However, my team feels that treating names as case-sensitive should not be a bug.
Here are our reasons:
* Some names (especially surnames) has upper-cases in the middle. In such instances,  case-sensitivity of names has meaning in the real world and should be reflected in code as well. For eg, [Colin McRae](https://en.wikipedia.org/wiki/Colin_McRae) should be treated differently from someone who (although however unlikely, is entirely possible) is named Colin Mcrae.
* Usages of adjectives in names. For eg, a person could be called "George the Third" or [Cléo de Mérode](https://en.wikipedia.org/wiki/Cl%C3%A9o_de_M%C3%A9rode). In such cases, the presence of these **non-capitalized** adjectives in their names differentiates them from people that (although however unlikely, is entirely possible) are actually named "George The Third" or "Cléo De Mérode".
* Give full control and more freedom to the user

In conclusion, as there are proper usages of case sensitivity when it comes to names in the real world, we feel that treating names as case-sensitive in code should not be considered a bug despite what the module website suggests. Perhaps we could also make it explicit in our user guide that names are case-sensitive.

Are these valid enough reasons to say that treating names as case-sensitive is not a bug?

**Comment by damithc** [comment]
@ekweirui If this bug is reported in the PE, and if you can convince the bug reporter that the target user of your specific application is better off when the name matching is case sensitive, then it will not be counted as a feature flaw. 

It also depends on where the matching happens.
Consider a context of searching an address book: If the user searches for `colin`, would the user be better off if `Colin` was omitted from the search results? Even if they are two people, would it be better to include `Colin` in the result and let the user decide?

Regardless, making the current behavior clear to the user (e.g., mention it in the UG) is a good thing.

**Comment by teoyuqi** [comment]
> Regardless, making the current behavior clear to the user (e.g., mention it in the UG) is a good thing.

Suppose we decide to make it clearer to the user by updating the application's error message. E.g., when the user requests to edit a non-existent `colin` contact instead of `Colin`, we now explicitly display a note to remind users that names are case sensitive (on top of the current implementation of only indicating that `colin` does not exist). 

Would that be considered a feature modification?

**Comment by damithc** [comment]
> Suppose we decide to make it clearer to the user by updating the application's error message. E.g., when the user requests to edit a non-existent `colin` contact instead of `Colin`, we now explicitly display a note to remind users that names are case sensitive (on top of the current implementation of only indicating that `colin` does not exist).

This is a good way to handle this case.

> 
> Would that be considered a feature modification?

Yes. So, update the UG with current behavior and also mention that more flexible name matching is planned for a future release.
